og_ui/src/EventSubscriber/OgUiEventSubscriber.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Drupal\og_ui\EventSubscriber;
+
+use Drupal\og\Event\PermissionEventInterface;
+use Drupal\og\OgRoleInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscribers for Organic Groups.
+ */
+class OgUiEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      PermissionEventInterface::EVENT_NAME => [['provideDefaultOgPermissions']],
+    ];
+  }
+
+  /**
+   * Provides default OG permissions.
+   *
+   * @param \Drupal\og\Event\PermissionEventInterface $event
+   *   The OG permission event.
+   */
+  public function provideDefaultOgPermissions(PermissionEventInterface $event) {
+    $event->setPermissions([
+      'subscribe' => [
+        'title' => t('Subscribe to group'),
+        'description' => t('Allow non-members to request membership to a group (approval required).'),
+        'roles' => [OgRoleInterface::ANONYMOUS],
+        'default roles' => [OgRoleInterface::ANONYMOUS],
+      ],
```

Comment:
And this is the actual code that runs. This is almost identical to `og_ui_og_permission()` except that we don't return a list of permissions, but set it with `$event->setPermissions()`. We also have a bunch of other methods available to inspect and alter the existing permissions (e.g. `$event->getPermission()`, `$event->deletePermission()` etc).

We can also get some metadata, like the group type and bundle, with `$event->getEntityTypeId()` and `$event->getBundleId()`.